CHANGELOG.md

```diff
@@ -6,12 +6,24 @@ The Azure MCP Server updates automatically by default whenever a new release com
 
 ### Features Added
 
+- Introduced `BaseAzureResourceService` class that uses Azure Resource Graph queries for improved performance and reduced API calls when retrieving Azure SQL resources. This provides faster and more efficient resource discovery compared to direct ARM API calls.
+
 ### Breaking Changes
 
 ### Bugs Fixed
 
+- Fixed KQL string escaping in Workbooks service queries to prevent potential injection issues and improve query reliability.
+- Fixed SQL service test assertions to use case-insensitive string comparisons for resource type validation.
+- Fixed HttpClient service test assertions to properly validate NoProxy collection handling instead of expecting a single string value.
+
 ### Other Changes
 
+- Refactored SQL service implementation to use Azure Resource Graph queries instead of direct ARM API calls, significantly improving performance and reducing the number of API requests required for SQL resource operations.
+- Removed dependency on `Azure.ResourceManager.Sql` package by migrating to Azure Resource Graph queries, reducing package size and improving startup performance.
+- Added `BaseAzureResourceService` abstract class that provides common Resource Graph query functionality with tenant caching and consistent error handling for Azure services.
+- Enhanced `BaseAzureService` with `EscapeKqlString` method for safe KQL query construction across all Azure services.
+- Updated development documentation to include guidance on choosing between `BaseAzureResourceService` and `BaseAzureService` based on service requirements.
```

Comment:
Let's reduce the text in changelog.  You can put under a main bullet with a couple of points.  This is too verbose for a changelog.